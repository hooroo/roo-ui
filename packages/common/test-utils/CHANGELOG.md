# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
