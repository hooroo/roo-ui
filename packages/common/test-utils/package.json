{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
