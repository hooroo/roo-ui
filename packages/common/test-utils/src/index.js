export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
