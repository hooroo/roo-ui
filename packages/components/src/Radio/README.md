# Radio

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { Box, Label, Radio } from '@roo-ui/components';

export default (
  <Box textAlign="left" is="form">
    <Label textStyle="text">
      <Radio name="example" /> Uno
    </Label>

    <Label textStyle="text">
      <Radio name="example" /> Dos
    </Label>

    <Label textStyle="text">
      <Radio name="example" /> Tres
    </Label>

    <Label textStyle="text">
      <Radio disabled name="example" /> Quattro
    </Label>
  </Box>
);
```

## Properties

| Name       | Description           | Type   | Default | Required? |
|:-----------|:----------------------|:-------|:--------|:----------|
| `checked`  | sets checked state    | `bool` | false   | -         |
| `disabled` | disables radio button | `bool` | false   | -         |

Standard [radio attributes](https://www.w3schools.com/jsref/dom_obj_radio.asp) will be passed down to the `<input />`.
