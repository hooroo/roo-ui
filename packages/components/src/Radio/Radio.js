import React from 'react';
import styled from 'styled-components';
import { themeGet } from 'styled-system';
import { rem } from 'polished';
import tag from 'clean-tag';

const Radio = props => (
  <RadioWrapper>
    <RadioInput type="radio" {...props} />
    <RadioIcon />
  </RadioWrapper>
);

const RadioWrapper = styled(tag.div)`
  display: inline-block;
  position: relative;
`;

const RadioInput = styled.input`
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;

  &:focus {
    box-shadow: none;
  }
`;

const RadioIcon = styled(tag.div)`
  display: inline-block;
  position: relative;
  vertical-align: middle;
  margin-bottom: ${rem('2px')};
  margin-right: ${themeGet('space.2')};
  height: ${rem('20px')};
  width: ${rem('20px')};
  background-color: ${themeGet('colors.white')};
  border: ${themeGet('borders.2')};
  border-color: ${themeGet('colors.grey.2')};
  border-radius: 50%;
  z-index: 1;

  input:checked + &,
  input:focus + &,
  input:hover + & {
    border-color: ${themeGet('colors.brand.secondary')};
    transition: ${themeGet('transitions.default')}
  }

  input:checked + & {
    &:after {
      position: absolute;
      display: block;
      content: '';
      top: ${themeGet('space.1')};
      left: ${themeGet('space.1')};
      height: ${themeGet('space.2')};
      width: ${themeGet('space.2')};
      border-radius: 50%;
      background-color: black;
    }
  }

  input:disabled + & {
    background-color: ${themeGet('colors.grey.2')};
    border-color: ${themeGet('colors.grey.2')};
  }
`;

export default Radio;
