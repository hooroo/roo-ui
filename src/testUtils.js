import React from 'react';
import { mount, shallow } from 'enzyme';

export const mountWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return mount(themedTree);
};

export const shallowWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return shallow(themedTree);
};
