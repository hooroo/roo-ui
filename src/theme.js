import { rem } from 'polished';
import range from 'lodash/range';

import * as icons from './defaultIcons';

const namedColors = {
  white: '#FFFFFF',
  brightSun: '#FBCB3B',
  red: '#E00E00',
  maroon: '#870000',
  blue: '#8DE2E0',
  lightBlue: '#E7F7F7',
  orange: '#ED710B',
  lightOrange: '#FCEBCD',
  green: '#35A509',
  lightGreen: '#DEF2DE',
};

const brand = {
  primary: '#E40000',
  secondary: '#8DE2E0',
  tertiary: '#D20000',
  quaternary: '#F9F3E9',
};

const greys = {
  charcoal: '#323232',
  steel: '#666666',
  alto: '#DADADA',
  porcelain: '#F4F5F6',
  dusty: '#999999',
  mineShaft: '#222222',
};

const ui = {
  error: namedColors.orange,
  errorBackground: namedColors.lightOrange,
  info: namedColors.blue,
  infoBackground: namedColors.lightBlue,
  link: namedColors.red,
  linkHover: namedColors.maroon,
  success: namedColors.green,
  successBackground: namedColors.lightGreen,
};

const colors = {
  brand,
  greys,
  ui,
  ...namedColors,
};

const alertStyles = {
  default: { bg: 'greys.porcelain' },
  error: {
    bg: 'ui.errorBackground',
    icon: {
      name: 'warning',
      color: 'ui.error',
    },
  },
  info: {
    bg: 'ui.infoBackground',
    icon: {
      name: 'info',
      color: 'greys.charcoal',
    },
  },
  success: {
    bg: 'ui.successBackground',
    icon: {
      name: 'checkCircle',
      color: 'ui.success',
    },
  },
};

const borders = [0, '1px solid', '2px solid', '3px solid'];

const breakpoints = ['40rem', '52rem', '64rem'];

const buttons = {
  default: {
    color: 'white',
    backgroundColor: 'greys.charcoal',
  },
  primary: {
    color: 'white',
    backgroundColor: 'brand.primary',
  },
};

const fontFamily = 'Ciutadella, sans-serif';

const fontSizes = {
  xs: rem('13px'),
  sm: rem('15px'),
  base: rem('16px'),
  lg: rem('18px'),
  xl: rem('22px'),
  '2xl': rem('26px'),
  '3xl': rem('36px'),
  '4xl': rem('46px'),
};

const fontWeights = {
  normal: 500,
  bold: 600,
};

const letterSpacings = {
  normal: '0',
  wide: '0.125em',
};

const lineHeights = {
  tight: 1,
  normal: 1.35,
  loose: 1.5,
};

const maxWidths = {
  default: rem('1280px'),
};

const mediaQueries = breakpoints.map(
  n => `@media screen and (min-width: ${n})`,
);

const opacity = {
  disabled: 0.7,
};

const radii = {
  none: 0,
  default: '4px',
  rounded: '1000px',
};

const shadows = {
  none: 0,
  default: '0 1px 1px 0 rgba(0, 0, 0, 0.1)',
  heavy: '0 2px 2px 0 rgba(0, 0, 0, 0.1)',
  focus: `0 0 2px 2px ${colors.greys.dusty};`,
};

const space = range(0, 129, 4).map(i => rem(`${i}px`));

const textStyles = {
  caps: {
    textTransform: 'uppercase',
    fontWeight: fontWeights.bold,
  },

  h1: {
    fontSize: fontSizes['3xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h2: {
    fontSize: fontSizes['2xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h3: {
    fontSize: fontSizes.xl,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h4: {
    fontSize: fontSizes.lg,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h5: {
    fontSize: fontSizes.base,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h6: {
    fontSize: fontSizes.sm,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  label: {
    fontSize: fontSizes.sm,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.bold,
    color: colors.greys.charcoal,
  },
  text: {
    fontSize: fontSizes.base,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.normal,
  },
};

const transitions = {
  default: '200ms ease-in',
};

export default {
  alertStyles,
  borders,
  breakpoints,
  buttons,
  colors,
  fontFamily,
  fontSizes,
  fontWeights,
  icons,
  letterSpacings,
  lineHeights,
  maxWidths,
  mediaQueries,
  opacity,
  radii,
  shadows,
  space,
  textStyles,
  transitions,
};
