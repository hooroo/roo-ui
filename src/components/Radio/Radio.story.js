import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';

import Radio from './';
import { Box, Label } from '../';
import README from './README.md';

storiesOf('Components/Radio', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <Box textAlign="left" as="form">
      <Label textStyle="text">
        <Radio name="example" /> Uno
      </Label>

      <Label textStyle="text">
        <Radio name="example" /> Dos
      </Label>

      <Label textStyle="text">
        <Radio name="example" /> Tres
      </Label>

      <Label textStyle="text">
        <Radio disabled name="example" /> Quattro (disabled)
      </Label>
    </Box>
  ));
