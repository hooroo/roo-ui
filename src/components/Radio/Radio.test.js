import React from 'react';
import theme from 'theme';
import { axe, mountWithTheme } from 'testUtils';
import { renderIntoDocument, isDOMComponent } from 'react-dom/test-utils';

import Radio from './';

describe('<Radio />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithTheme(<Radio />, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has no accessibility errors', async () => {
    wrapper = mountWithTheme(
      <label htmlFor="radio">
        Label
        <Radio />
      </label>,
      theme,
    );
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  it('can handle a ref and pass it into a DOM component', () => {
    const ref = React.createRef();

    renderIntoDocument(<Radio ref={ref} />);
    expect(isDOMComponent(ref.current)).toEqual(true);
  });

  describe('when checked', () => {
    beforeEach(() => {
      wrapper = mountWithTheme(<Radio defaultChecked />, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when disabled', () => {
    beforeEach(() => {
      wrapper = mountWithTheme(<Radio disabled />, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
