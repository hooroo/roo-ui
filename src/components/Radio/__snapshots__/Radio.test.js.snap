// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> renders correctly 1`] = `
.emotion-4 {
  display: inline-block;
  position: relative;
}

.emotion-0 {
  opacity: 0;
  position: absolute;
  z-index: 0;
}

.emotion-0:focus {
  box-shadow: none;
}

.emotion-2 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  margin-bottom: 0.125rem;
  height: 1.25rem;
  width: 1.25rem;
  border-radius: 50%;
  z-index: 1;
}

input:checked + .emotion-2:after {
  position: absolute;
  display: block;
  content: '';
  border-radius: 50%;
}

<Radio>
  <RadioWrapper>
    <div
      className="emotion-4 emotion-5"
    >
      <RadioInput
        type="radio"
      >
        <input
          className="emotion-0 emotion-1"
          type="radio"
        />
      </RadioInput>
      <RadioIcon>
        <div
          className="emotion-2 emotion-3"
        />
      </RadioIcon>
    </div>
  </RadioWrapper>
</Radio>
`;

exports[`<Radio /> when checked renders correctly 1`] = `
.emotion-4 {
  display: inline-block;
  position: relative;
}

.emotion-0 {
  opacity: 0;
  position: absolute;
  z-index: 0;
}

.emotion-0:focus {
  box-shadow: none;
}

.emotion-2 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  margin-bottom: 0.125rem;
  height: 1.25rem;
  width: 1.25rem;
  border-radius: 50%;
  z-index: 1;
}

input:checked + .emotion-2:after {
  position: absolute;
  display: block;
  content: '';
  border-radius: 50%;
}

<Radio
  defaultChecked={true}
>
  <RadioWrapper>
    <div
      className="emotion-4 emotion-5"
    >
      <RadioInput
        defaultChecked={true}
        type="radio"
      >
        <input
          className="emotion-0 emotion-1"
          defaultChecked={true}
          type="radio"
        />
      </RadioInput>
      <RadioIcon>
        <div
          className="emotion-2 emotion-3"
        />
      </RadioIcon>
    </div>
  </RadioWrapper>
</Radio>
`;

exports[`<Radio /> when disabled renders correctly 1`] = `
.emotion-4 {
  display: inline-block;
  position: relative;
}

.emotion-0 {
  opacity: 0;
  position: absolute;
  z-index: 0;
}

.emotion-0:focus {
  box-shadow: none;
}

.emotion-2 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  margin-bottom: 0.125rem;
  height: 1.25rem;
  width: 1.25rem;
  border-radius: 50%;
  z-index: 1;
}

input:checked + .emotion-2:after {
  position: absolute;
  display: block;
  content: '';
  border-radius: 50%;
}

<Radio
  disabled={true}
>
  <RadioWrapper>
    <div
      className="emotion-4 emotion-5"
    >
      <RadioInput
        disabled={true}
        type="radio"
      >
        <input
          className="emotion-0 emotion-1"
          disabled={true}
          type="radio"
        />
      </RadioInput>
      <RadioIcon>
        <div
          className="emotion-2 emotion-3"
        />
      </RadioIcon>
    </div>
  </RadioWrapper>
</Radio>
`;
